e2e/nomostest/nt.go

```diff
@@ -625,28 +662,32 @@ func (nt *NT) portForwardGitServer() *portforwarder.PortForwarder {
 		}
 		// re-init all repos
 		InitGitRepos(nt, allRepos...)
-		// attempt to recover by re-pushing the local repo states
-		for _, remoteRepo := range nt.RemoteRepositories {
-			if err := remoteRepo.PushAllBranches(); err != nil {
+		for _, repo := range allRepos {
+			// construct remoteURL with provided port. Calling LocalPort would lead to deadlock
+			remoteURL, err := provider.RemoteURLWithPort(newPort, repo.String())
+			if err != nil {
+				nt.T.Fatal(err)
+			}
+			if err := nt.RemoteRepositories[repo].PushAllBranches(remoteURL); err != nil {
 				nt.T.Fatal(err)
 			}
 		}
 		prevPodName = podName
 	}
-	return nt.newPortForwarder(
+	nt.T.Cleanup(func() {
```

Comment:
comment: // clear PortFowarder after each test, so it gets recreated in test setup.

Is this actually necessary or does test setup replace it anyway?

Reply:
It's not strictly necessary but I thought it would be good form. At this point in the `Cleanup` stack the `PortForwarder` will have been stopped, so it can no longer be used. Added a comment